tests/bytecode.rs

```diff
@@ -0,0 +1,27 @@
+#[cfg(test)]
+mod tests {
+    use core::panic;
+
+    use yaiwr::{instruction::Instruction, Calc};
+
+    #[test]
+    fn bytecode() {
+        let ast = Calc::from_str("1*2").unwrap();
+        let bytecode = &mut vec![];
+        let bytecode = Calc::to_bytecode(ast, bytecode);
+        assert_eq!(bytecode.len(), 3);
+
+        match bytecode[0] {
```

Comment:
I *think* you can make this much more concise with a single pattern match over a slice?

https://www.reddit.com/r/rust/comments/qiap4i/array_pattern_matching/

Reply:
rewritten as slice pattern match :point_right:   https://github.com/softdevteam/pavel.yaiwr/pull/10/commits/bbe7318c59b283fe809ed81e9aa45136fe5e9e74